File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/hashx/fnv1a64.go
package hashx

// Fnv1a64 computes the fnv1a64 checksum over raw bytes.
func Fnv1a64(data []byte) uint64 {
	state := uint64(14695981039346656037)
	for _, b := range data {
		state ^= uint64(b)
		state *= 1099511628211
	}
	return state
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/hash/fnv1a64.cpp
#include "freight/hashes.h"

namespace freight {

// fnv1a64 over raw bytes.
uint64_t hash_fnv1a64(const std::string& data) {
  uint64_t state = 14695981039346656037ULL;
  for (size_t i = 0; i < data.size(); ++i) {
    state ^= static_cast<uint8_t>(data[i]);
    state *= 1099511628211ULL;
  }
  return state;
}

}  // namespace freight
